test/integration/setting_js_test.rb

```diff
@@ -29,4 +29,77 @@ class SettingJSTest < IntegrationTestWithJavascript
     assert_index_page(settings_path, "Settings", false, true, false)
     assert page.has_link?("My Pretty Setting Label", :href => "#category_label_test_settings_tab")
   end
+
+  test "general headers values" do
+    visit settings_path
+
+    assert_text 'Name'
+    assert_text 'Value'
+    assert_text 'Description'
+  end
+
+  test "edit single arrays values" do
+    visit settings_path
+    click_button(['#http_proxy_except_list'])
+
+    input = find('textarea#setting-textarea-http_proxy_except_list')
+    assert_empty input.value
+
+    input.fill_in with: 'testArr'
+    find('button[data-ouia-component-id="submit-edit-btn"]').click
```

Comment:
FYI: I didn't point to every instance in your changes that could be improved. Ideally we use `click_button` where we can and also enhance the assertions in this file.